pkg/controller/account/account_controller.go

```diff
@@ -245,212 +181,296 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 		// see if in creating for longer then default wait time
 		if accountCreatingTooLong(currentAcctInstance) {
 			r.setStatusFailed(reqLogger, currentAcctInstance, fmt.Sprintf("Creation pending for longer then %d minutes", utils.WaitTime))
+			// TODO(efried): Return here?
```

Comment:
Is this awaiting anything?  Or has this been worked out?

Reply:
If there needs to be a return there (I think there does, but would like maintainer scrutiny on it, please) then I think it should be done in a separate PR, as the goal of this one is pure refactor, with zero logic change.

AFAIU, this PR as a whole is waiting for me to get back around to it and add unit testing for these code paths. Which obviously inflates the effort by an order of magnitude, so it may be a hot minute before I get to it.